videos/videos.js

```diff
@@ -34,19 +35,30 @@ AWS.config.update({
 
 const s3 = new AWS.S3();
 
-
-
 // Check if S3 file exists
-router.get("/check-for-file", (req, res) => {
-  const { fileName } = req.body;
-
-  s3.headObject(params, function (err, metadata) {
-    if (err && err.code === "NotFound") {
-      // Handle no object on cloud here
-    } else {
-      s3.getSignedUrl("getObject", params, callback);
+router.get("/single/check-for-file", async (req, res) => {
+    const { fileName } = req.body;
+    console.log(req.body);
+    const params = {
+      Bucket: process.env.AWS_BUCKET,
+      Key: fileName,
+    };
+    console.log(params);
+    try {
+      s3.headObject(params, function (err, metadata) {
+        if (err && err.code === "NotFound") {
+          // Handle no object on cloud here
+          console.log(err);
+        } else {
+          s3.getSignedUrl("getObject", params, (err, data) => {
```

Comment:
That's a nice callback function.  I am going to have to go over some of the documentation to learn more about these AWS methods.